src/radical/pilot/pmgr/launching/base.py

```diff
@@ -265,14 +266,26 @@ def _cancel_pilots(self, pids):
             # nothing to do
             return
 
-        # record time of request, so that forceful termination can happen
-        # after a certain delay
-        now = time.time()
         with self._pilots_lock:
             for pid in pids:
-                if pid in self._pilots:
-                    self._log.debug('update cancel req: %s %s', pid, now)
-                    self._pilots[pid]['pilot']['cancel_requested'] = now
+
+                self._log.debug('cancel pilot %s', pid)
+                if pid not in self._pilots:
+                    self._log.warn('cannot cancel unknown pilot %s', pid)
+                    continue
+
+                pilot = self._pilots[pid]
+                lname = pilot['launcher']
+
+                if lname not in self._launchers:
+                    self._log.warn('invalid pilot launcher name: %s', lname)
+                    continue
+
+                launcher = self._launchers[lname]
+                try:
+                    launcher.cancel_pilots([pid])
```

Comment:
currently launcher classes have `_cancel_pilots` methods, better to keep them without starting `_`, since they are public methods